client/src/views/home/DeviceJoinOrganizationModal.vue

```diff
@@ -39,8 +203,236 @@ function parseInvitationLink(link: string): ClaimDeviceLink {
-const waitingForHost = ref(true);
+const waitingForHost = ref(false);
+
+interface Title {
+  title: string,
+  subtitle?: string,
+}
+
+const titles = new Map<DeviceJoinOrganizationStep, Title>([[
+  DeviceJoinOrganizationStep.Information, {
+    title: t('ClaimDeviceModal.titles.claimDevice'),
+  }], [
+  DeviceJoinOrganizationStep.GetHostSasCode, {
+    title: t('ClaimDeviceModal.titles.getCode'),
+    subtitle: t('ClaimDeviceModal.subtitles.getCode'),
+  }], [
+  DeviceJoinOrganizationStep.ProvideGuestCode, {
+    title: t('ClaimDeviceModal.titles.provideCode'),
+    subtitle: t('ClaimDeviceModal.subtitles.provideCode'),
+  }], [
+  DeviceJoinOrganizationStep.Password, {
+    title: t('ClaimDeviceModal.titles.password'), subtitle: t('ClaimDeviceModal.subtitles.password'),
+  }], [
+  DeviceJoinOrganizationStep.Finish, {
+    title: t('ClaimDeviceModal.titles.done', {org: ''}),
+  }],
+]);
+
+function selectHostSas(_code: string | null): void {
+  nextStep();
+}
+
+function getNextButtonText(): string {
+  if (pageStep.value === DeviceJoinOrganizationStep.Information) {
+    return t('ClaimDeviceModal.buttons.understand');
+  } else if (pageStep.value === DeviceJoinOrganizationStep.Password) {
+    return t('ClaimDeviceModal.buttons.password');
+  } else if (pageStep.value === DeviceJoinOrganizationStep.Finish) {
+    return t('ClaimDeviceModal.buttons.login');
+  }
+
+  return '';
+}
+
+const nextButtonIsVisible = computed(() => {
+  return (
+    pageStep.value === DeviceJoinOrganizationStep.Information
+    || pageStep.value === DeviceJoinOrganizationStep.Password
+    || pageStep.value === DeviceJoinOrganizationStep.Finish
+  );
+});
+const canGoForward = computed(() => {
+  if (pageStep.value === DeviceJoinOrganizationStep.Password) {
+    return passwordPage.value.areFieldsCorrect();
+  }
+  return true;
+});
+
+function cancelModal(): Promise<boolean> {
+  if (pageStep.value === DeviceJoinOrganizationStep.Finish) {
+    return modalController.dismiss({ device: newDevice, password: passwordPage.value.password }, ModalResultCode.Confirm);
+  } else {
+    return modalController.dismiss(null, ModalResultCode.Cancel);
+  }
+}
+
+function mockCreateDevice(): AvailableDevice {
+  const humanHandle = 'Louis Dark';
+  const deviceName = 'myDevice';
+  console.log(
+    `Creating device ${claimDeviceLink?.org}, user ${humanHandle}, device ${deviceName}, backend ${props.invitationLink}`,
+  );
+  const device: AvailableDevice = {
+    organizationId: claimDeviceLink?.org || '',
+    humanHandle: humanHandle,
+    deviceLabel: deviceName,
+    keyFilePath: 'key_file_path',
+    deviceId: 'device1@device1',
+    slug: 'slug1',
+    ty: {tag: 'Password'},
+  };
+
+  return device;
+}
+
+function mockSaveDeviceWithPassword(_device: AvailableDevice | null, _password: string): void {
+  console.log(`Saving device with password ${passwordPage.value.password}`);
+}
+
+function nextStep(): void {
+  newDevice = mockCreateDevice();
```

Comment:
Only at the right step, don't create the newDevice here. It should only be created right before the mockSaveDeviceWithPassword.

Reply:
Done